client/src/App.jsx

```diff
@@ -4,23 +4,75 @@ import Calendar from './Calendar';
 import Guests from './Guests';
 import Reserve from './Reserve';
 
-
 class App extends React.Component {
   constructor(props) {
     super(props);
+
+    this.state = {
+      guestDropdownVisibility: "hidden",
```

Comment:
Looks like you are passing many of these states more than 1 level down. Is there ways we can refactor to have fewer of them in App.jsx?